sources/api/netdog/src/cli/write_primary_interface_status.rs

```diff
@@ -0,0 +1,116 @@
+use super::{error, primary_interface_name, Result};
+use crate::dns::DnsSettings;
+use crate::networkd_status::NetworkDInterfaceStatus;
+use crate::{CURRENT_IP, PRIMARY_SYSCTL_CONF, SYSCTL_MARKER_FILE, SYSTEMD_SYSCTL};
+use argh::FromArgs;
+use snafu::{ensure, ResultExt};
+use std::fmt::Write;
+use std::fs;
+use std::net::IpAddr;
+use std::path::Path;
+use std::process::Command;
+
+#[derive(FromArgs, PartialEq, Debug)]
+#[argh(subcommand, name = "write-primary-interface-status")]
+/// Updates the various files needed when responding to events
+pub(crate) struct WritePrimaryInterfaceStatusArgs {}
+
+/// Updates the various files needed when responding to events
+pub(crate) fn run() -> Result<()> {
+    let primary_interface = primary_interface_name()?;
+
+    let primary_link_status = NetworkDInterfaceStatus::new(primary_interface.clone())
+        .context(error::NetworkDInterfaceStatusSnafu {})?;
+
+    // Write out current IP
+    let primary_ip = &primary_link_status
+        .primary_address()
+        .context(error::PrimaryInterfaceAddressSnafu {})?;
+    write_current_ip(primary_ip)?;
+
+    // Write out resolv.conf
+    write_resolv_conf(&primary_link_status)?;
+
+    // If we haven't already, set and apply default sysctls for the primary network
+    // interface
+    if !Path::exists(Path::new(PRIMARY_SYSCTL_CONF)) {
+        write_interface_sysctl(primary_interface, PRIMARY_SYSCTL_CONF)?;
+    };
+
+    // Execute `systemd-sysctl` with our configuration file to set the sysctls
+    if !Path::exists(Path::new(SYSCTL_MARKER_FILE)) {
+        let systemd_sysctl_result = Command::new(SYSTEMD_SYSCTL)
+            .arg(PRIMARY_SYSCTL_CONF)
+            .output()
+            .context(error::SystemdSysctlExecutionSnafu)?;
+        ensure!(
+            systemd_sysctl_result.status.success(),
+            error::FailedSystemdSysctlSnafu {
+                stderr: String::from_utf8_lossy(&systemd_sysctl_result.stderr)
+            }
+        );
+
+        fs::write(SYSCTL_MARKER_FILE, "").unwrap_or_else(|e| {
+            eprintln!(
+                "Failed to create marker file {}, netdog may attempt to set sysctls again: {}",
+                SYSCTL_MARKER_FILE, e
+            )
+        });
+    }
```

Comment:
Can this be refactored into a shared function? It seems like a lot of copy/paste code from `install.rs`.

Reply:
It is a copy paste from `install.rs` since it will essentially replace that file entirely. I figured it was short lived so factoring out to a function may be focusing too much on a temporary condition. Nonetheless, I can factor it out to a function.